### ISSUE TITLE
Hotbar render issue + Chest (only with shaders) render issue - 1.21 Fabric

### ISSUE THREAD

**Issue by ObsyTek** [FabianMPunkt/Nautilus3D#25]
### Game Version

Minecraft 1.21 (1.21/fabric/Fabric)

### List all installed mods

- Fabulously Optimized Modpack - Version 6.0.0-beta.6 for 1.21 (detailed below). Nothing else added outside Nautilus3D and Shaders

Dynamic FPS - 3.6.3
e4mc - 5.1.0
Controlify - 2.0.0-beta.14+1.21-fabric
M.R.U - 1.0.2+1.21-fabric
FerriteCore - 7.0.0
Fabrishot - 1.14.0
ModernFix - 5.19.0+mc1.21
Mod Menu - 11.0.1
Mod Menu Helper.zip
FastQuit - 3.0.0+1.20.6
OptiGUI - 2.3.0-beta.2+1.21
Sodium Extra - mc1.21-0.5.7
Better Mount HUD - 1.2.4
Entity Culling - 1.6.6
MixinTrace - 1.1.1+1.17
Sodium - mc1.21-0.5.11
ImmediatelyFast - 1.2.18+1.21-fabric
More Culling - 0.26.0
[EMF] Entity Model Features - 2.1.3
Polytone - 1.21-2.2.10
Main Menu Credits - 1.2.0
Animatica - 0.6.1+1.21
Paginated Advancements & Custom Frames - 2.4.2
Iris Shaders - 1.7.3+1.21
CreativeCore - 2.12.9
Debugify - 1.21+1.0
Fabric API - 0.100.8+1.21
Enhanced Block Entities - 0.10.1+1.21
Lithium - mc1.21-0.12.7
Puzzle - 1.6.3+1.21-Puzzle
Your Options Shall Be Respected (YOSBR) - 0.1.2
Language Reload - 1.6.1+1.21
Remove Reloading Screen - 5.0.1+mc1.21-fabric
Reese's Sodium Options - mc1.21-1.7.3
Cloth Config API - 15.0.128+fabric
Model Gap Fix - 1.21-1.6
Capes - 1.5.4+1.21
FabricSkyboxes - mc1.21-0.7.4
FabricSkyBoxes Interop - mc1.21-1.4.0
No Chat Reports - Fabric-1.21-v2.8.0
Zoomify - 2.14.0+1.21
Chat Reporting Helper - 4.19
YetAnotherConfigLib - 3.5.0+1.21-fabric
Translations for Sodium - v3.25.0
Continuity - 3.0.0-beta.5+1.21
Fast Better Grass - 1.20.3
More Chat History - 1.3.1
Cubes Without Borders - 2.1.1+1.21
[ETF] Entity Texture Features - 6.1.3
Indium - 1.0.34+mc1.21
Fabric Language Kotlin - 1.11.0+kotlin.2.0.0

### List of all loaded packs

- Nautilus3D-V2.1.zip
- Mod Menu Helper.zip
- SodiumTranslations.zip
- Mods Fabric
- Default

### Describe your issue

On this first screen, items in the hotbar can be replaced by a brown square or basically just desappear. (here I have in order one wooden axe, 11 planks, 2 sticks and 1 chest. You can only see the planks). It does not happen in inventory. Also the breaking effect doesn't work on chests.
![Capture d'écran 2024-09-08 011822](https://github.com/user-attachments/assets/2a1f0424-3d98-4fe0-b3c4-864922b71598)

It's even more weird when using shaders. (here it's Solas Shader but I also tested with Make-Up - Ultrafast that you recommend on the shader page. As you can see, the chest that's supposed to be in my hand weirdly float beside me. If I turn on myself it does not turn with me.
![Capture d'écran 2024-09-08 011910](https://github.com/user-attachments/assets/d28920b9-b9d4-499f-b78e-49371fc27523)

**Comment by ObsyTek** [comment]
I did some more digging (basically disabling all mods ans enabling them again one by one) and got 2 new infos : 
- Chest problems seems to be caused by [EMF] Entity Model Features - 2.1.3. I tried updating it to it's latest version for 1.21 and problem remain. Should it be reported to the creator of this mod too ?
- While doing my tests, I could not recreate the hotbar problem at all ? But did not change anything between the screen I shared and the end of the tests, all the same mods enabled, no more no less. I'm confused.

**Comment by FabianMPunkt** [comment]
Thanks for reporting, I suspect EMF to be the culprit here. What version of EMF did you use after updating? Version 2.2.4 works for me ;)
EMF is know to be a bit spotty at times. Could also be another mod messing with EMF somehow. I'll have to see if i can reproduce the issue using the same Modpack.
The Hotbar problem must be something else, since Nautilus3D doesn't change anything about the wooden axe (or any tools for that matter)

**Comment by ObsyTek** [comment]
When updating I did try with 2.2.4 (and with the latest update for ETF). I'll try doing some tests on my side too, with the other mods and EMF now that I know it's, at least partially, the culprit x). Currently I just disabled EMF since I could do without.

**Comment by ObsyTek** [comment]
I did more digging ! Detailed process so you know how I ended up with my conclusion but might be long. TLDR / conclusion at the end.

Step 1 : 
- I tried with a new fresh instance of Fabulously Optimized 1.21.1 and only added Nautilus3D and Shaders.
- Launched a new save, break a chest.
At first, the only remaining problem was the breaking effect texture still not appearing but after breaking the chest, it was in my hand as supposed.

Step 2 : 
- Left the game then came back on the already created save, immediatly without changing anything
- Broke chest again
The chest was not in my hand anymore but as shown on the second pic of my post. So it doesn't do it the first time but does when coming back in the save. (and the breaking effect texture still missing)

Step 3 : 
- Disable mods from the modpack one by one, testing in between each the save, stopping when I only had Fabric API, Iris (with Sodium necessary), Complementary shader, EMF (with ETF necessary)
- Broke chest again
Still no breaking effect texture and chest in hand floating.

Step 4 : 
- Disable Shader but keep Iris (and Sodium)
- Broke chest again
Still no breaking effect texture


TLDR / Conclusion : 
- Breaking effect texture missing is due to EMF alone
- Floating chest in hand is due to EMF + active shader (at least both Complementary and Make-Up - Ultrafast, only after joining an already made save.

**Comment by ObsyTek** [comment]
Installed the latest version, did not change any settings and added complementary shader and nautilus only. Still got the same problem. First launch of the save I only get the missing breaking effect texture when breaking a chest (first GIF). After leaving and launching the same file again it does what the second GIFs below shows. (had to put links, it was too heavy to add diretcly as files)
[Breaking texture missing](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExOGZiMHhkMjhwZnFoMmtuOWdmZTRsNXNxNnN3cmd0Mnltb3hkaHRtciZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/VHZs1kSRItjioQjNw7/giphy.gif)
[Chest in hand rendering issue](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNzVpZ2Ywb3VlcHNzdXJ2a3ZianJzNHljemp4cXhoaHRwZm96NW9qOSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/8FnYEo8HsiiG5McOLp/giphy-downsized-large.gif)

**Comment by FabianMPunkt** [comment]
Huh now that's weird.. I only tried Solas & MakeUp shader with the modpack. Complementary works fine with only Sodium & Iris - I'll give it a try on Fabulously Optimized. I honestly have no Idea how it manages to screw up the handmodel so bad.
Can you try and download Nautilus3D again to make sure you didn't end up with a corrupted download🙈?

I did find a way to fix the particle effects. The game uses the oak texture by default and because the chests themselves are entity-models, they don't allow for custom break-particles. (Normal entities don't really need that lol).

**Comment by ObsyTek** [comment]
Every time I tried it was with a new fresh install of everything, never used previous files ! I download everything directly from Modrinth App, and use the latest versions, and try not to touch any settings. GIFs I send were with a fresh install of the latest version of Fabulously Optimized, same for Nautilus3D and Complementary. 
Also as I said previously, the only mod that actually seem to be part of the problem is EMF when coupled with shaders. No EMF or no shader = no problem !

**Comment by FabianMPunkt** [comment]
Okay i'll take a closer look at it again. I remember the GLSL Shaders mod from way back in the day had settings for hadmodel rendering that always caused problems like this. No idea how it works nowadays in Iris though.

**Comment by FabianMPunkt** [comment]
Took a closer look at this today and could finally reliably reproduce the problem. It has to do with how the textures are assigned to the .jem models. It believe this to be a bug in EMF but i can work around it, which could also result in potentoal performance increase. It will take some time as i have to remake almost all the models.

**Comment by FabianMPunkt** [comment]
Good news! I found a workaround and i managed to fix the break particles.
The problem was infact related to the textures link in the chest model.
![2024-09-18_11 39 05](https://github.com/user-attachments/assets/a2bf6b3d-c6ff-4ba3-a1b1-45af9e0c2889)
![2024-09-18_11 39 08](https://github.com/user-attachments/assets/28127343-2b55-47e9-b0dc-9a095b07648f)

 I also redesigned the entire chest model so it is made up of much fewer elements, resulting in a noticable performance increase.
Fix will be in the next update!
